neuraxle/metaopt/trial.py

```diff
@@ -50,32 +55,224 @@ def __init__(
+
+    def set_success(self):
+        """
+        Set trial status to success.
+        """
+        self.status = TRIAL_STATUS.SUCCESS
+
+    def update_final_trial_status(self):
+        """
+        Set trial status to success.
+        """
+        success = True
+        for validation_split in self.validation_splits:
+            if not validation_split.is_success():
+                success = False
+
+        if success:
+            self.status = TRIAL_STATUS.SUCCESS
+        else:
+            self.status = TRIAL_STATUS.FAILED
+
+    def set_failed(self, error: Exception):
+        """
+        Set failed trial with exception.
+
+        :param error: catched exception
+        :return:
+        """
+        self.status = TRIAL_STATUS.FAILED
+        self.error = str(error)
+        self.error_traceback = traceback.format_exc()
+
+    def _get_trial_hash(self, hp_dict: Dict):
+        """
+        Hash hyperparams with md5 to create a trial hash.
+
+        :param hp_dict: hyperparams dict
+        :return:
+        """
+        current_hyperparameters_hash = hashlib.md5(str.encode(str(hp_dict))).hexdigest()
+        return current_hyperparameters_hash
+
+    def to_json(self):
+        return {
+            'status': self.status.value,
+            'hyperparams': self.hyperparams.to_flat_as_dict_primitive(),
+            'validation_splits': [v.to_json() for v in self.validation_splits],
+            'error': self.error,
+            'error_traceback': self.error_traceback,
+            'start_time': self.start_time.strftime(TRIAL_DATETIME_STR_FORMAT) if self.start_time is not None else '',
+            'end_time': self.end_time.strftime(TRIAL_DATETIME_STR_FORMAT) if self.end_time is not None else ''
+        }
+
+    @staticmethod
+    def from_json(trial_json) -> 'Trial':
+        return Trial(
+            status=TRIAL_STATUS(trial_json['status']),
+            hyperparams=HyperparameterSamples(trial_json['hyperparams']),
+            validation_splits=[
+                TrialSplit.from_json(validation_split_json)
+                for validation_split_json in trial_json['validation_splits']
+            ],
+            error=trial_json['error'],
+            error_traceback=trial_json['error_traceback'],
+            start_time=datetime.datetime.strptime(trial_json['start_time'], TRIAL_DATETIME_STR_FORMAT),
+            end_time=datetime.datetime.strptime(trial_json['start_time'], TRIAL_DATETIME_STR_FORMAT)
+        )
+
+    def __enter__(self):
+        """
+        Start trial, and set the trial status to PLANNED.
+        """
+        self.start_time = datetime.datetime.now()
+        self.status = TRIAL_STATUS.STARTED
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """
+        Stop trial, and save end time.
+
+        :param exc_type:
+        :param exc_val:
+        :param exc_tb:
+        :return:
+        """
+        self.end_time = datetime.datetime.now()
+        del self.pipeline
```

Comment:
Would the deletion of the pipeline here suggests that the pipeline might need to be passed at another moment than with `set_pipeline`?

Reply:
it wasn't even used anymore lol. just had to delete the method.